lib/services/elasticsearch.js

```diff
@@ -473,6 +482,10 @@ class ElasticSearch extends Service {
     assertNoRouting(esRequest);
     assertWellFormedRefresh(this.config.apiVersion, esRequest);
 
+    if (!this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)) {
```

Comment:
not necessary for replace, as we check the document existence just in the lines below.
If the document exists, the index/collection necessary exists too.